docs/pipeline.md

```diff
@@ -0,0 +1,19 @@
+# Pipeline Mock-up
+Create an infrastructure for home-labs for a very simple web app.
+
+## The objective is to gain practice on the following items:
+
+- Version Control Systems - Git, GitHub
+- CI/CD Tools - Jenkins X, Argo CD
+- Containerization and Orchestration - Kubernetes
+- Container Registries - Docker Hub
+- Infrastructure as Code (IaC) - Helm, Terraform
+- Monitoring and Logging - Prometheus, Grafana
+- Service Mesh - Istio
+- GitOps Tools - Flux, Argo CD
```

Comment:
Argo is mentioned twice, also at CI/CD tools

Although GitOps is part of CD, it can definitely be its own category.

Perhaps we can mention Azure DevOps Pipelines, Tekton and Gitlab CI for CI/CD solutions, for example

Reply:
Yes, I drafted this list so that it could be open for discussion and yes, Argo CD was mentioned there because as you say I understand it can play both roles. As for the cloud technologies I initially left them out so that it could as much as possible an home-lab setup but I certainly don't know enough about the subject and I'll gladly follow your lead.